packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-outbound-section.tsx

```diff
@@ -0,0 +1,469 @@
+
+  const onItemsSelected = async () => {
+    itemsToAdd.length &&
+      (await addOutboundItem(
+        {
+          items: itemsToAdd.map((variantId) => ({
+            // TODO: Line item will not be present at this stage. Do we skip this?
+            // id: variantId,
+            variant_id: variantId,
+            quantity: 1,
+          })),
+        },
+        {
+          onError: (error) => {
+            toast.error(error.message)
+          },
+        }
+      ))
+
+    for (const itemToRemove of itemsToRemove) {
+      const actionId = previewOutboundItems
+        .find((i) => i.variant_id === itemToRemove)
+        ?.actions?.find((a) => a.action === "ITEM_ADD")?.id
+
+      if (actionId) {
+        await removeOutboundItem(actionId, {
+          onError: (error) => {
+            toast.error(error.message)
+          },
+        })
+      }
+    }
+
+    setIsOpen("outbound-items", false)
+  }
+
+  // TODO: implement outbound shipping
+  const { mutateAsync: updateClaimRequest, isPending: isUpdating } = {} // useUpdateClaim(claim.id, order.id)
+  const onLocationChange = async (selectedLocationId?: string | null) => {
+    await updateClaimRequest({ location_id: selectedLocationId })
+  }
+
+  const onShippingOptionChange = async (selectedOptionId: string) => {
+    const promises = preview.shipping_methods
+      .map((s) => s.actions?.find((a) => a.action === "SHIPPING_ADD")?.id)
+      .filter(Boolean)
+      .map(deleteInboundShipping)
+
+    await Promise.all(promises)
+
+    await addInboundShipping(
+      { shipping_option_id: selectedOptionId },
+      {
+        onError: (error) => {
+          toast.error(error.message)
+        },
+      }
+    )
+  }
+
+  const showLevelsWarning = useMemo(() => {
+    if (!locationId) {
+      return false
+    }
+
+    const allItemsHaveLocation = outboundItems
+      .map((i) => {
+        const item = variantItemMap.get(i.variant_id)
+        if (!item?.variant_id || !item?.variant) {
+          return true
+        }
+
+        if (!item.variant.manage_inventory) {
+          return true
+        }
+
+        return inventoryMap[item.variant_id]?.find(
+          (l) => l.location_id === locationId
+        )
+      })
+      .every(Boolean)
+
+    return !allItemsHaveLocation
+  }, [outboundItems, inventoryMap, locationId])
+
+  useEffect(() => {
+    // TODO: Implement this correctly
+    const getInventoryMap = async () => {
+      const ret: Record<string, InventoryLevelDTO[]> = {}
+
+      if (!outboundItems.length) {
+        return ret
+      }
+
+      ;(
+        await Promise.all(
+          outboundItems.map(async (_i) => {
+            const item = variantItemMap.get(_i.variant_id)!
+
+            if (!item?.variant_id || !item.variant?.product) {
+              return undefined
+            }
+
+            // TODO: use variant list endpoint instead
+            return await sdk.admin.product.retrieveVariant(
```

Comment:
question: This is part of the PR – do we want to change it now?

Reply:
just pushed 🤞🏻